### PR TITLE
Automatic stitching in SANSILLAutoProcess

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

An automatically stitched result is also produced by the SANSILLAutoProcess algorithm. When the automatic stitching succeed, a workspace named `OutputWorkspace` + "_stitched"` is created.

**To test:**

3rd test of `SANSILLAutoProcessTest`:
```python
config['default.facility'] = 'ILL'
config['default.instrument'] = 'D16'
config.appendDataSearchSubDir('ILL/D16/')                                       
SANSILLAutoProcess(
    SampleRuns="3674,3677,3680",
    BeamRuns="3587",
    DefaultMaskFile="side_mask",
    MaskFiles="beam_mask, side_mask, side_mask",
    TransmissionBeamRuns="3587",
    OutputWorkspace="ws",
    ContainerTransmissionRuns="3673",
    SampleTransmissionRuns="3671",
    ContainerRuns="3676,3679,3682",
    AbsorberRuns="3598,3604,3654",
    ThetaDependent=False,
    WaterCrossSection=0.87,
    SampleThickness=0.2,
    BeamRadius=1
)
```
Check for workspace `ws_stitched`.

Part of #26906 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py [hunk 1]
```diff
@@ -328,6 +328,19 @@ def PyExec(self):
             else:
                 self.log().information('Skipping empty token run.')
 
+        for output in outputs:
+            ConvertToPointData(InputWorkspace=output,
+                               OutputWorkspace=output)
+        if len(outputs) > 1:
```
Comment (gvardany) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py:334]: Stitching should happen only if the output type is I(Q), so one-dimensional.

File: Testing/SystemTests/tests/analysis/SANSILLAutoProcessTest.py [hunk 2]
```diff
@@ -63,6 +65,11 @@ def runTest(self):
                 OutputWorkspace='iq_s' + str(i + 1)
             )
 
+            try:
```
Comment (gvardany) [inline Testing/SystemTests/tests/analysis/SANSILLAutoProcessTest.py:68]: You should rather update the references to cover also the stitched output.

**Review by gvardany — COMMENTED** [review]
Few comments, and make sure this works also for the wedges.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py (hunk 1 above)
Comment (gui-co) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py]: I am now checking that OutputType is I(Q).

File: Testing/SystemTests/tests/analysis/SANSILLAutoProcessTest.py (hunk 2 above)
Comment (gui-co) [reply Testing/SystemTests/tests/analysis/SANSILLAutoProcessTest.py]: I changed the reference files when needed and removed the try-catch blocks.

**Pull request merged**